### ISSUE TITLE
Cyclonedx parsing not understanding components.

### ISSUE THREAD

**Issue by agonzalez-plume** [evryfs/sbom-dependency-submission-action#131]
Running cdxgen to generate SBOMs for a variety of projects.  The format that is output doesn't include the `dependencies` section or include components in the `metadata` section, so this submission action only wants to submit a blank snapshot. 

Adding a patch to traverse through the `components` section and use those as dependencies should fix the problem.

Format of the output is... a shortened version of the output.

```"bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "serialNumber": "urn:uuid:7f5748c9-5062-4dfe-b0f0-60f829088f33",
  "version": 1,
  "metadata": {
    "timestamp": "2022-10-18T16:06:46.044Z",
    "tools": [
      {
        "vendor": "AppThreat",
        "name": "cdxgen",
        "version": "4.0.41"
      }
    ],
    "authors": [
      {
        "name": "Team AppThreat",
        "email": "cloud@appthreat.com"
      }
    ]
  },
  "components": [
    {
      "group": "software.amazon.awssdk",
      "name": "aws-core",
      "version": "2.10.53",
      "description": "",
      "scope": "required",
      "hashes": [
        {
          "alg": "SHA-1",
          "content": "c51d535a2dc9d4326cea471ebe2997c1eca3ce17"
        }
      ],
      "licenses": [],
      "purl": "pkg:maven/software.amazon.awssdk/aws-core@2.10.53",
      "type": "library",
      "bom-ref": "pkg:maven/software.amazon.awssdk/aws-core@2.10.53"
    },
    {
      "group": "software.amazon.eventstream",
      "name": "eventstream",
      "version": "1.0.1",
      "description": "",
      "scope": "required",
      "hashes": [
        {
          "alg": "SHA-1",
          "content": "6ff8649dffc5190366ada897ba8525a836297784"
        }
      ],
      "licenses": [],
      "purl": "pkg:maven/software.amazon.eventstream/eventstream@1.0.1",
      "type": "library",
      "bom-ref": "pkg:maven/software.amazon.eventstream/eventstream@1.0.1"
    }
  ],
  "externalReferences": [
    {
      "type": "other",
      "url": ".",
      "comment": "Base path"
    },
    {
      "type": "other",
      "url": "home/runner/work/sample/sample",
      "comment": "Package file"
    }
  ]
}
```

snapshot 
```
{
    "detector": {
        "name": "cdxgen",
        "version": "4.0.41",
        "url": "https://"
    },
    "version": 0,
    "job": {
        "correlator": "sbom_jobsbom-sample-cdxgen.json",
        "id": "3275067353"
    },
    "sha": "12406834c66a0b248df6d00b504529402b0a1268",
    "ref": "refs/heads/Dependencies",
    "scanned": "2022-10-18T16:06:46.044Z",
    "manifests": {
        "sbom-sample-cdxgen.json": {
            "resolved": {},
            "name": "sbom-sample-cdxgen.json"
        }
    }
}
```

**Comment by agonzalez-plume** [comment]
@davidkarlsen Maybe I'm missing something simple here.

In looking at the samples being used for tests. This one: https://github.com/evryfs/sbom-dependency-submission-action/blob/main/__tests__/data/base_ubuntu_syft_packages.json seems to match my output.

https://github.com/evryfs/sbom-dependency-submission-action/blob/main/__tests__/data/dropwizard-1.3.15-sbom.json has a 'dependencies' section

https://github.com/evryfs/sbom-dependency-submission-action/blob/main/__tests__/data/keycloak-10.0.2-sbom.json has a 'dependencies' section

https://github.com/evryfs/sbom-dependency-submission-action/blob/main/__tests__/data/valid-bom-1.4.json has a 'dependencies' section

but https://github.com/evryfs/sbom-dependency-submission-action/blob/main/__tests__/data/base_ubuntu_syft_packages.json  doesn't.

So maybe something else is going on.  Did the base_ubuntu_syft_packages.json work right in the tests?

**Comment by agonzalez-plume** [comment]
My experience in this depgraph stuff comes from a different graph building I am using for gradle projects. A snippet of the very long snapshot output shows:

```{
    "detector": {
        "name": "mikepenz/gradle-dependency-submission",
        "url": "https://github.com/mikepenz/gradle-dependency-submission",
        "version": "0.7.0"
    },
    "version": 0,
    "job": {
        "correlator": "build-:sample:DebugCompileClasspath",
        "id": "3133396008"
    },
    "sha": "7240145a38aa04ad6ff886490f984dd0185097d2",
    "ref": "refs/heads/Dependencies",
    "scanned": "2022-09-27T06:27:00.527Z",
    "manifests": {
        "sample": {
            "resolved": {
                "pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk8@1.6.10": {
                    "package_url": "pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk8@1.6.10",
                    "relationship": "direct",
                    "dependencies": [
                        "pkg:maven/org.jetbrains.kotlin/kotlin-stdlib@1.6.21",
                        "pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk7@1.6.21"
                    ]
                },
                "pkg:maven/com.google.firebase/firebase-bom@30.1.0": {
                    "package_url": "pkg:maven/com.google.firebase/firebase-bom@30.1.0",
                    "relationship": "direct",
                    "dependencies": [
                        "pkg:maven/com.google.firebase/firebase-perf@20.1.0",
                        "pkg:maven/com.google.firebase/firebase-core@21.0.0",
                        "pkg:maven/com.google.firebase/firebase-firestore@24.1.2",
                        "pkg:maven/com.google.firebase/firebase-config@21.1.0",
```

Whereas, this one when I am giving it a SBOM only shows very little, with me thinking the parsing wasn't working correctly.  These are two different projects so don't compare the two directly, but I'm giving some context that I expected to see much more in the snapshot notice so I felt like it wasn't parsing correctly. When i went into the code, I thought I saw why, but not certain now that you have a test case similar to what I thought the bug was.

```
Notice: {
    "detector": {
        "name": "cdxgen",
        "version": "4.0.41",
        "url": "https://"
    },
    "version": 0,
    "job": {
        "correlator": "sbom_jobsbom-sample-cdxgen.json",
        "id": "3285387512"
    },
    "sha": "aeb750ae7106391036210e7a00af6c1bdb82b330",
    "ref": "refs/heads/Dependencies",
    "scanned": "2022-10-19T22:22:22.206Z",
    "manifests": {
        "sbom-sample-cdxgen.json": {
            "resolved": {},
            "name": "sbom-sample-cdxgen.json"
        }
    }
}
Notice: Snapshot successfully created at 2022-10-19T22:22:23.[15]
```

**Comment by agonzalez-plume** [comment]
I downloaded the repo and started changing the tests. I added the syft sbom file for ubuntu and it isn't creating a snapshot for that correctly either.  The problem is with the `dependencies` section not being in these boms

The problem lies in line main.ts:90. 
```
if (!deps.length && sbom.dependencies?.length && sbom.components) {
```
sbom.dependencies is null so the if statement fails

Changing this line to seems to fix it without problems.
```
if (!deps.length && sbom.components) {
```
I submitted a PR, but maybe there something you know about the simple change that could affect something else.
